automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/user/SignInPage.scala

```diff
@@ -67,17 +68,17 @@ class SignInPage(val baseUrl: String)(implicit webDriver: WebDriver) extends Fir
 class GoogleSignInPopup(implicit webDriver: WebDriver) extends WebBrowser with WebBrowserUtil {
 
   def awaitLoaded(): GoogleSignInPopup = {
-
-    val popupTrial = Try {
-      await condition (id("identifierLink").findElement.exists(_.isDisplayed) || id("identifierId").findElement.exists(_.isEnabled),10)
-    }
-
-    popupTrial match {
+    Try {
+      await condition (id("identifierLink").findElement.exists(_.isDisplayed)
+        || id("identifierId").findElement.exists(_.isEnabled)
+        || id("Email").findElement.exists(_.isDisplayed), 10) // One Account All of Google popup
+    } match {
       case Success(_) =>
-        val chooseAccountLink = find(id("identifierLink")).isDefined
-        if (chooseAccountLink) {
-          click on (id("identifierLink"))
-          await visible id("identifierId")
+        find(id("identifierLink")).isDefined match {
```

Comment:
I don't think the use of `match` is more readable than the `if`, especially since you're not doing anything in the `false` case.

You could try something like this:
```
find(id("identifierLink")) foreach { link =>
  click on link
  await visible id("identifierId")
}
```
It reads a little funny because `find` returns only a single element in an `Option`, but `foreach` is a perfectly valid function of `Option`... like `map` but returning `Unit` instead of transforming the value.

Reply:
I like this